A32NX/html_ui/Pages/A32NX_Utils/NXApi.js

```diff
@@ -14,6 +14,21 @@ class NXApi {
             });
     }
 
+    static getAirport(icao) {
+        if (!icao) {
+            throw new Error("No ICAO provided");
+        }
+
+        return fetch(`${NXApi.url}/api/v1/airport/${icao}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error("Request failed with status code " + response.status);
```

Comment:
**question**: All other code simply throws the response. Why did you choose not to do that?

Reply:
Because beheh change like that on before PR

Reply:
Yeah, I think the current version is fine. I'm not convinced throwing the response is a good pattern. A code that calls `getAirport` should not be expected to inspect the response and look at status codes to figure out what went wrong.